### PR TITLE
Prompt user for password when OSF_PASSWORD is not set.

### LINKED ISSUE #30: Interactive password prompt
If a username is given but `OSF_PASSWORD` isn't set, prompt the user for it.

### PR DESCRIPTION
**Pull request by La0**
Fixes #30

### CODE REVIEW

**Comment by betatim** [comment]
Looks good. Can you think of a way that we can test this?

**Comment by La0** [comment]
We can [mock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) `getpass.getpass` and just check that the password value is the one from the unittest

File: osfclient/tests/test_cli.py [hunk 1]
```diff
@@ -80,3 +80,13 @@ def simple_config(key):
     expected = ('specify a project ID via the command line, configuration '
                 'file or environment variable')
     assert expected in e.value.args[0]
+
+def test_password_prompt():
+
```
Comment (betatim) [inline osfclient/tests/test_cli.py]: Can you add a blank line above `def test_password_prompt` and delete the blank after it? Just to make the `pep8` gods happy :smile:

**Comment by betatim** [comment]
LGTM after the nitpick on formatting!

**Comment by betatim** [comment]
Thanks a lot! Some virtual 🍕 in time for lunch.

**Comment by betatim** [comment]
Just added you to the collaborators for this repo. We try and run this using https://github.com/dib-lab/osf-cli/issues/10. This means everyone who has code merged can in the future merge code.

With great power comes great responsibility :)

Welcome on board!

**Pull request merged**